integrations/gitlab/gitlab_integration/gitlab_service.py

```diff
@@ -44,26 +56,29 @@ def __init__(
         self.app_host = app_host
         self.group_mapping = group_mapping
 
-    def _is_exists(self, group: RESTObject) -> bool:
+    def _does_webhook_exist_for_group(self, group: RESTObject) -> bool:
         for hook in group.hooks.list(iterator=True):
             if hook.url == f"{self.app_host}/integration/hook/{group.get_id()}":
                 return True
         return False
 
-    def _create_group_webhook(self, group: RESTObject) -> None:
-        logger.info(f"Creating webhook for {group.get_id()}")
+    def _create_group_webhook(
+        self, group: RESTObject, events: list[str] | None
+    ) -> None:
+        webhook_events_to_listen_to: list[str]
+        if not events:
+            webhook_events_to_listen_to = self.all_events_in_webhook
+        else:
+            webhook_events_to_listen_to = events
+
+        logger.info(
+            f"Creating webhook for {group.get_id()} with events: {webhook_events_to_listen_to}"
+        )
+
         resp = group.hooks.create(
             {
                 "url": f"{self.app_host}/integration/hook/{group.get_id()}",
-                "push_events": True,
-                "merge_requests_events": True,
-                "issues_events": True,
-                "job_events": True,
-                "pipeline_events": True,
-                "releases_events": True,
-                "tag_push_events": True,
-                "subgroup_events": True,
-                "confidential_issues_events": True,
+                **{event_name: True for event_name in webhook_events_to_listen_to},
             }
```

Comment:
per to our discussion, lets verify that if events are not passed that they are actually set to false in gitlab ( in case that their default if not set is `True` )

Reply:
fixed